xds/internal/client/load/store.go

```diff
@@ -20,63 +20,121 @@ package load
 import (
 	"sync"
 	"sync/atomic"
+	"time"
 )
 
 const negativeOneUInt64 = ^uint64(0)
 
-// A pair of cluster and service name. The same cluster can be used by multiple
-// services, and one service can use multiple clusters. So we need a pair with
-// both name to accurately indicate where the load belongs.
-type storeKey struct {
-	cluster string
-	service string
-}
-
 // Store keeps the loads for multiple clusters and services to be reported via
-// LRS. It is safe for concurrent use.
+// LRS. It contains loads to reported to one LRS server. Create multiple stores
+// for multiple servers.
+//
+// It is safe for concurrent use.
 type Store struct {
-	mu       sync.RWMutex
-	clusters map[storeKey]*PerClusterStore
+	// mu only protects the map (2 layers). The read/write to *perClusterStore
+	// doesn't need to hold the mu.
+	mu sync.RWMutex
+	// clusters is a map with cluster name as the key. The second layer is a map
+	// with service name as the key. Each value (perClusterStore) contains data
+	// for a (cluster, service) pair.
+	//
+	// Note that new entries are added to this map, but never removed. This is
```

Comment:
Is there a way to avoid this? Just like how the users of the `xdsClient` need to cancel their watches, they are also required to call the `cancel` function returned from `ReportLoad`. Could we use that somehow?